docs/Analyzer.md

```diff
@@ -16,21 +16,27 @@ and `@nowarn` annotation for warning suppression.
+| `explicitGenerics`         | warning       | Enables [`@explicitGenerics`](https://github.com/AVSystem/scala-commons/blob/master/commons-core/src/main/scala/com/avsystem/commons/annotation/explicitGenerics.scala) checking                                                  |
+| `discardedMonixTask`       | warning       | Makes sure that expressions evaluating to Monix `Task`s are not accidentally discarded by conversion to `Unit`                                                                                                                    |
+| `throwableObjects`         | warning       | Makes sure that objects are never used as `Throwable`s (unless they have stack traces disabled)                                                                                                                                   |
+| `constantDeclarations`     | off           | Checks if constants are always declared as `final val`s with `UpperCamelCase` and no explicit type annotation for literal values                                                                                                  |
+| `basePackage`              | warning       | Checks if all sources are within the specified base package                                                                                                                                                                       |
+| `catchThrowable`           | warning       | Makes sure that code does not catch `Throwable` directly, which can hide critical errors like `OutOfMemoryError`                                                                                                                  |
+| `finalValueClasses`        | warning       | Makes sure that value classes are marked final.                                                                                                                                                                                   |
+| `finalCaseClasses`         | warning       | Makes sure that case classes are marked final. Does not affect inner classes due to SI-4440                                                                                                                                       |
+| `implicitParamDefaults`    | warning       | Makes sure that default values are not defined for implicit parameters                                                                                                                                                            |
+| `implicitValueClasses`     | warning       | Makes sure that implicit classes extend `AnyVal` (when applicable). Available options: `top-level-only` (default) - only checks implicit classes in top-level objects; `all` - checks both top-level and nested implicit classes. |
```

Comment:
NIT:
 it's not clear how to pass the options to this flag - try formatting them like https://docs.scala-lang.org/overviews/compiler-options/index.html